infra/modules/azure_naming_convention/variables.tf

```diff
@@ -1,42 +1,38 @@
-variable "environment" {
-  type = object({
+variable "environments" {
+  type = list(object({
     prefix          = string
     env_short       = string
     location        = string
     domain          = optional(string)
     app_name        = string
-    instance_number = string
-  })
+    instance_number = number
+  }))
 
   validation {
-    condition     = length(var.environment.prefix) == 2
-    error_message = "The variable \"prefix\" must contain 2 characters"
+    condition     = alltrue([for env in var.environments : length(env.prefix) == 2])
+    error_message = "Each environment's prefix must contain exactly 2 characters."
   }
 
   validation {
-    condition     = contains(["d", "u", "p"], var.environment.env_short)
-    error_message = "Allowed values for \"env_short\" are \"d\", \"u\", \"p\"."
+    condition     = alltrue([for env in var.environments : length(env.app_name) > 1])
+    error_message = "The variable \"app_name\" must contain at least 2 characters"
   }
 
   validation {
-    condition     = contains(["italynorth", "westeurope", "germanywestcentral", "spaincentral", "northeurope"], var.environment.location)
-    error_message = "Allowed values for \"location\" are \"italynorth\" \"westeurope\", \"germanywestcentral\", \"spaincentral\", \"northeurope\"."
+    condition     = alltrue([for env in var.environments : contains(["d", "u", "p"], env.env_short)])
+    error_message = "Allowed values for env_short are 'd', 'u', 'p'."
   }
 
   validation {
-    condition     = var.environment.domain == null ? true : length(replace(var.environment.domain, "-", "")) >= 2
```

Comment:
Why validations on `domain` and `app_name` have been removed?

Reply:
To be honest, it seemed unnecessary to me, as the domain is optional and the app_name is not. 
However, I am not entirely sure why there was a two-character limit in place

Reply:
Just to be sure a team choose a meaningful domain name (at least 2 chars) and a non-empty app name